pkg/vsphere/disk/disk_manager.go

```diff
@@ -421,3 +430,53 @@ func (m *Manager) devicePathByURI(op trace.Operation, datastoreURI *object.Datas
 
 	return sysPath, nil
 }
+
+func (m *Manager) InUse(op trace.Operation, config *VirtualDiskConfig) (*vm.VirtualMachine, error) {
+	defer trace.End(trace.Begin(""))
+
+	var vms []mo.VirtualMachine
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	parent, err := m.vm.ResourcePool(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create view of VirtualMachine objects under the VCH's resource pool
+	v, err := m.vMngr.CreateContainerView(ctx, parent.Reference(), []string{"VirtualMachine"}, true)
```

Comment:
Can we hang the view off `storage.Config` or similar and create it on-demand if not present (or not functional)? That would avoid three API calls per invocation of this method.